LiteDB/Client/Mapper/BsonMapper.GetEntityMapper.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+
+namespace LiteDB;
+
+public partial class BsonMapper
+{
+    /// <summary>
+    /// Mapping cache between Class/BsonDocument
+    /// </summary>
+    private readonly ConcurrentDictionary<Type, EntityMapper> _entities = new();
+
+    /// <summary>
+    /// Get property mapper between typed .NET class and BsonDocument - Cache results
+    /// </summary>
+    internal EntityMapper GetEntityMapper(Type type)
+    {
+        if (_entities.TryGetValue(type, out EntityMapper mapper))
+        {
+            return mapper;
+        }
+
+        using var cts = new CancellationTokenSource();
```

Comment:
I'm honestly not sure what was the point of all this refactoring, as the only thing I had in mind is to replace the Dictionary with ConcurrentDictionary, everything else here looks like an unnecessary overengineering.